src/js/common/schemaform/SelectiveArrayField.jsx

```diff
@@ -0,0 +1,203 @@
+import _ from 'lodash/fp';
+import classNames from 'classnames';
+import Scroll from 'react-scroll';
+
+
+import {
+  toIdSchema,
+  getDefaultFormState,
+  deepEquals
+} from 'react-jsonschema-form/lib/utils';
+
+const Element = Scroll.Element;
+const scroller = Scroll.scroller;
+
+
+class SelectiveArrayField extends React.Component {
+  // This fills in an empty item in the array if it has minItems set
+  // so that schema validation runs against the fields in the first item
+  // in the array. This shouldn't be necessary, but there's a fix in rjsf
+  // that has not been released yet
+  componentDidMount = () => {
+    const { schema, formData = [], registry } = this.props;
+    if (schema.minItems > 0 && formData.length === 0) {
+      this.props.onChange(Array(schema.minItems).fill(
+        getDefaultFormState(schema.additionalItems, undefined, registry.definitions)
+      ));
+    }
+  }
+
+  shouldComponentUpdate = (nextProps, nextState) => {
+    return !deepEquals(this.props, nextProps) || nextState !== this.state;
+  }
+
+  onItemChange = (indexToChange, value) => {
+    const newItems = _.set(indexToChange, value, this.props.formData || []);
+    this.props.onChange(newItems);
+  }
+
+  getItemSchema = (index, selectedFields) => {
+    const schema = this.props.schema;
+    const itemSchema = schema.items.length > index
+      ? schema.items[index]
+      : schema.additionalItems;
+
+    const properties = _.pick(selectedFields, itemSchema.properties);
+    const required = _.intersection(selectedFields, itemSchema.required);
+
+    return _.assign(itemSchema, {
+      properties,
+      required
+    });
+  }
+
+  scrollToTop = () => {
+    setTimeout(() => {
+      scroller.scrollTo(`topOfTable_${this.props.idSchema.$id}`, {
+        duration: 500,
+        delay: 0,
+        smooth: true,
+        offset: -60
+      });
+    }, 100);
+  }
+
+  scrollToRow = (id) => {
+    setTimeout(() => {
+      scroller.scrollTo(`table_${id}`, {
+        duration: 500,
+        delay: 0,
+        smooth: true,
+        offset: 0
+      });
+    }, 100);
+  }
+
+  render() {
+    const {
+      uiSchema,
+      errorSchema,
+      idSchema,
+      formData,
+      disabled,
+      readonly,
+      registry,
+      formContext,
+      onBlur,
+      schema
+    } = this.props;
+    const definitions = registry.definitions;
+    const { TitleField, SchemaField } = registry.fields;
+
+    const title = uiSchema['ui:title'] || schema.title;
+    const hideTitle = !!_.get(['ui:options', 'hideTitle'], uiSchema);
+    const description = uiSchema['ui:description'];
+    const textDescription = typeof description === 'string' ? description : null;
+    const DescriptionField = typeof description === 'function'
+      ? uiSchema['ui:description']
+      : null;
+    const hasTitleOrDescription = (!!title && !hideTitle) || !!description;
+    const selectedFields = uiSchema['ui:selectedFields'] || Object.keys(schema.properties);
```

Comment:
Right, since we moved to a single object for formData, both pages will modify the same children array. So we don't have to worry about data merge issues later.